presto-native-execution/presto_cpp/main/operators/ShuffleInterface.h

```diff
@@ -19,6 +19,10 @@ namespace facebook::presto::operators {
 
 class ShuffleInterface {
  public:
+  /// Setup the memory pool to be used by the shuffle client.
+  /// This must be the very first call into the shuffle service.
+  virtual bool initializeMemoryManager(velox::memory::MemoryPool* pool) = 0;
```

Comment:
@beroyfb Based on recent discussion about shuffle, it seems like we ned a ShuffleManager with a factor method. Roughly like this:

```
std::unique_ptr<ShuffleInterface> create(const ShuffleInfo&, memory::MemoryPool* poo)
```

Reply:
I agree. Do you want those change as a part of this PR though?

Reply:
It would be nice to iterate on this PR to come up with the design that works for the use cases we have.